src/Infrastructure.VS/LanguageDetection/SonarLanguageRecognizer.cs

```diff
@@ -112,5 +119,37 @@ private static bool IsTypeScriptDocument(IEnumerable<IContentType> contentTypes)
         {
             return contentTypes.Any(type => type.IsOfType("TypeScript"));
         }
+
+        public AnalysisLanguage? GetAnalysisLanguageFromExtension(string fileName)
+        {
+            var extension = GetNormalizedExtention(fileName);
+
+            if (JavascriptSupportedExtensions.Contains(extension)) { return AnalysisLanguage.Javascript; }
+            if (IsContentType(TypeScriptTypeName, extension)) { return AnalysisLanguage.TypeScript; }
+            if (IsContentType(CFamilyTypeName, extension)) { return AnalysisLanguage.CFamily; }
+            if (IsContentType(CSharpTypeName, extension) || IsContentType(BasicTypeName, extension)) { return AnalysisLanguage.RoslynFamily; }
+
+            return null;
+        }
+
+        private string GetNormalizedExtention(string fileName)
+        {
+            var extension = Path.GetExtension(fileName);
+
+            if (extension.Length > 0 && extension[0] == '.')
+            {
+                //remove the leading . on extension
+                extension = extension.Substring(1);
+            }
+            return extension.ToLowerInvariant();
+        }
+
+        private bool IsContentType(string typeName, string extension)
+        {
+            var contentType = contentTypeRegistryService.GetContentType(typeName);
+            var extensions = fileExtensionRegistryService.GetExtensionsForContentType(contentType);
```

Comment:
I saw in the API that `fileExtensionRegistryService` has a method GetContentTypeForExtension.
Wouldn't that simplify? https://docs.microsoft.com/en-us/dotnet/api/microsoft.visualstudio.utilities.ifileextensionregistryservice.getcontenttypeforextension?view=visualstudiosdk-2022#microsoft-visualstudio-utilities-ifileextensionregistryservice-getcontenttypeforextension(system-string)